src/formatter/flatten.ts

```diff
@@ -0,0 +1,37 @@
+import * as O from "fp-ts/Option";
+import { withoutUndefinedValues } from "@pagopa/ts-commons/lib/types";
+import { pipe } from "fp-ts/lib/function";
+import { toJsonObject } from "../utils/data";
+
+export const flattenField = <T>(input: T, fieldToFlat: keyof T): T =>
```

Comment:
```suggestion
export const flattenField = <T extends Record<string, unknown>>(input: T, fieldToFlat: keyof T): T =>
```

could be more explicit?

Reply:
I've used the same definition of other mappings :)
Maybe we can change generic type definition for all others formatters